datafusion/functions-aggregate/src/min_max/min_max_generic.rs

```diff
@@ -0,0 +1,491 @@
+    }
+}
+
+/// Helper trait for sliding min/max accumulators to avoid code duplication
+pub(crate) trait GenericSlidingMinMaxAccumulatorHelper:
+    Debug + Default + Send + Sync
+{
+    fn push(&mut self, row: OwnedRow);
+
+    fn pop(&mut self);
+
+    fn value(&self) -> Option<&OwnedRow>;
+}
+
+#[derive(Debug, Default)]
+pub(crate) struct GenericMovingMin(MovingMin<OwnedRow>);
+
+impl GenericSlidingMinMaxAccumulatorHelper for GenericMovingMin {
+    fn push(&mut self, row: OwnedRow) {
+        self.0.push(row);
+    }
+
+    fn pop(&mut self) {
+        self.0.pop();
+    }
+
+    fn value(&self) -> Option<&OwnedRow> {
+        self.0.min()
+    }
+}
+
+#[derive(Debug, Default)]
+pub(crate) struct GenericMovingMax(MovingMax<OwnedRow>);
+
+impl GenericSlidingMinMaxAccumulatorHelper for GenericMovingMax {
+    fn push(&mut self, row: OwnedRow) {
+        self.0.push(row);
+    }
+
+    fn pop(&mut self) {
+        self.0.pop();
+    }
+
+    fn value(&self) -> Option<&OwnedRow> {
+        self.0.max()
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct GenericSlidingMinMaxAccumulator<
+    MovingWindowHelper: GenericSlidingMinMaxAccumulatorHelper,
+> {
+    /// Convert the columns to row so we can compare them
+    row_converter: RowConverter,
+
+    /// The current minimum value
+    current_value: Option<OwnedRow>,
+
+    moving_helper: MovingWindowHelper,
+
+    /// Null row to use for fast filtering
+    null_row: OwnedRow,
+}
+
+impl<MovingWindowHelper: GenericSlidingMinMaxAccumulatorHelper>
+    GenericSlidingMinMaxAccumulator<MovingWindowHelper>
+{
+    pub fn try_new(datatype: &DataType) -> Result<Self> {
+        let converter = RowConverter::new(vec![
+            (SortField::new_with_options(
+                datatype.clone(),
+                SortOptions {
+                    descending: false,
+                    nulls_first: true,
+                },
+            )),
+        ])?;
+
+        // Create a null row to use for filtering out nulls from the input
+        let null_row = {
+            let null_array = ScalarValue::try_from(datatype)?.to_array_of_size(1)?;
+
+            let rows = converter.convert_columns(&[null_array])?;
+
+            rows.row(0).owned()
+        };
+
+        Ok(Self {
+            row_converter: converter,
+            null_row,
+            current_value: None,
+            moving_helper: MovingWindowHelper::default(),
+        })
+    }
+}
+
+impl<MovingWindowHelper: GenericSlidingMinMaxAccumulatorHelper> Accumulator
+    for GenericSlidingMinMaxAccumulator<MovingWindowHelper>
+{
+    fn state(&mut self) -> Result<Vec<ScalarValue>> {
+        // Get the current value or null if no value has been seen
+        let min = self.current_value.as_ref().unwrap_or(&self.null_row);
+
+        Ok(vec![convert_row_to_scalar(&self.row_converter, min)?])
+    }
+
+    fn update_batch(&mut self, values: &[ArrayRef]) -> Result<()> {
+        // TODO - should assert getting only one column?
```

Comment:
I think it would be nice to either add the assert or remove the TODO as part of this PR